yaml/ruleset.go

```diff
@@ -137,12 +137,34 @@ func (r *Rules) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err == nil {
 		r.Branch = rules.Branch
 		r.Comment = rules.Comment
-		r.Event = rules.Event
 		r.Path = rules.Path
 		r.Repo = rules.Repo
 		r.Status = rules.Status
 		r.Tag = rules.Tag
 		r.Target = rules.Target
+
+		// account for users who use non-scoped pull_request event
+		events := []string{}
+
+		for _, e := range rules.Event {
+			switch e {
+			// backwards compatibility - pull_request = pull_request:opened + pull_request:synchronized
+			case constants.EventPull:
+				events = append(events,
+					constants.EventPull+":"+constants.ActionOpened,
+					constants.EventPull+":"+constants.ActionSynchronized)
+			// convert all and * to cover all configured actions
+			case constants.EventPull + ":" + "all", constants.EventPull + ":" + "*":
+				events = append(events,
+					constants.EventPull+":"+constants.ActionOpened,
+					constants.EventPull+":"+constants.ActionSynchronized,
+					constants.EventPull+":"+constants.ActionEdited)
```

Comment:
I'm not sure about introducing `pull_request:all` without including all of the possible github events. Adding more event types to this later will be another backwards compat issue because the definition of `:all` and `:*` will change. Can we include all known event types now instead of adding them piecemeal?

Reply:
i think the argument is that when the definition of "all" or "*" changes (by adding more events/actions whether that's likely or not), all existing pipelines using that would suddenly behave differently.

as implemented in this PR, we achieve backwards compatibility when the user just supplies `pull_request` as they did previously which is great. we might consider just dropping the "all"/"*" idea and only allow to explicitly list each action. arguably, the likelihood of folks actually wanting things to happen on _all_ events is low, and if they did, they still could.